### PR TITLE
Added script to generate docs

### LINKED ISSUE #159: Make docs fails after 130
The following error is reported after attempting to make docs - due to permission errors. 
```
ubuntu@ubuntu:~/DroneCore$ make docs
(cd /usr/local/include/dronecore \
        && mkdir -p docs \
        && doxygen /home/ubuntu/DroneCore/.doxygen \
        && /home/ubuntu/DroneCore/generate_markdown_from_doxygen_xml.py docs docs)
mkdir: cannot create directory ‘docs’: Permission denied
Makefile:74: recipe for target 'docs' failed
make: [docs] Error 1 (ignored)
ubuntu@ubuntu:~/DroneCore$ 
```
If you do `sudo` it works, but the docs are created under **/usr/local/include/dronecore/docs/markdown** (which I suggest is not the best way).

Probably our previous behaviour would be good here. As part of docs build create /install/include/ etc etc. In any case, should not force sudo to build docs, or build them in such an unlikely location.

### PR DESCRIPTION
**Pull request by julianoes**
This removes the `make docs` target and instead adds the script
generate_docs.sh to do a clean build, install locally and run doxygen,
and the generate_markdown_from_doxygen_xml script.

Fixes #159.

### CODE REVIEW

File: generate_docs.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# exit on any error
+set -e
+
+# This script will
+# 1. build and install the source,
+# 2. run doxygen and create html and xml docs,
+# 3. run script to generate markdown from xml
+
+# Get the directory where of this script.
```
Comment (hamishwillee) [inline generate_docs.sh]: Get script current directory.

("Get the directory where of this script." is confusing English)

File: generate_docs.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# We need to do a clean build, otherwise the INSTALL_PREFIX has no effect.
+if [ -e $source_dir/build/ ]; then
+    echo "Build directory ($install_prefix) already exists, you should do 'make clean' first."
+    exit 1
+fi
+
+# Check for leftover install artefacts.
+if [ -e $install_prefix ]; then
+    echo "Install directory ($install_prefix) already exists, you should clean it up first."
```
Comment (hamishwillee) [inline generate_docs.sh]: Perhaps terminate sentence as you did previously ... ", you should do 'make clean' first."

**Comment by hamishwillee** [comment]
Hi @julianoes Just FYI, also fixed up some bugs in APIRef discovered when building this.

File: generate_docs.sh (hunk 1 above)
Comment (julianoes) [reply generate_docs.sh]: Oops. Thx.

File: generate_docs.sh (hunk 2 above)
Comment (julianoes) [reply generate_docs.sh]: That's better, thx.

**Comment by julianoes** [comment]
@hamishwillee thanks for the review. I've added one comment fix commit and will merge it when CI is happy.

**Comment by julianoes** [comment]
Closes #167.

**Comment by hamishwillee** [comment]
Thank! FYI CI is happy now, so I'm going to merge it for you :-)

**Pull request merged**